pkg/controllers/routereflector/init.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2020 IBM Corporation All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routereflector
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/projectcalico/kube-controllers/pkg/config"
+	"github.com/projectcalico/kube-controllers/pkg/controllers/controller"
+	"github.com/projectcalico/kube-controllers/pkg/controllers/routereflector/datastores"
+	"github.com/projectcalico/kube-controllers/pkg/controllers/routereflector/topologies"
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	client "github.com/projectcalico/libcalico-go/lib/clientv3"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+type k8sNodeClient interface {
```

Comment:
I prefer to create my own interfaces, because easier to follow SOLID principles, code has the minimum knowledge of the dependency, that makes testing super easy (my other fetish :D).